File: Plateform_BackEnd/src/main/java/com/tourist/dto/SearchResultsDTO.java
package com.tourist.dto;

import java.util.List;

public class SearchResultsDTO {
    private List<EventDTO> events;
    private List<ExcursionDTO> excursions;
    private List<HotelDTO> hotels;

    public SearchResultsDTO(List<EventDTO> events, List<ExcursionDTO> excursions, List<HotelDTO> hotels) {
        this.events = events;
        this.excursions = excursions;
        this.hotels = hotels;
    }

    public List<EventDTO> getEvents() {
        return events;
    }

    public List<ExcursionDTO> getExcursions() {
        return excursions;
    }

    public List<HotelDTO> getHotels() {
        return hotels;
    }
}

File: Plateform_FrontEnd/src/app/dto/search-results.dto.ts
import {DtoEvent} from "./eventDTO/dto-event";
import {DtoExcursion} from "./excursionDTO/dto-excursion";
import {DtoHotel} from "./hotelDTO/dto-hotel";

export interface SearchResults {
  events: DtoEvent[];
  excursions: DtoExcursion[];
  hotels: DtoHotel[];
}
